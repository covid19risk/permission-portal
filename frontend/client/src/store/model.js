import { types, cast, onSnapshot } from 'mobx-state-tree'
import 'mobx-react-lite/batchingForReactDom'
import Logging from '../util/logging'

const User = types
  .model({
    isSignedIn: types.maybe(types.boolean),
    email: types.string,
    isAdmin: types.boolean,
    isSuperAdmin: types.boolean,
    disabled: types.boolean,
    prefix: types.maybe(types.string),
    imageBlob: types.maybeNull(types.string),
    firstName: types.string,
    lastName: types.string,
    organizationID: types.string,
    isFirstTimeUser: types.maybe(types.boolean),
  })
  .actions((self) => {
    const __update = (updates) => {
      Object.keys(updates).forEach((key) => {
        if (self.hasOwnProperty(key)) self[key] = updates[key] // eslint-disable-line no-prototype-builtins
      })
      Logging.log('Updated User:')
      Logging.log(self)
    }

    return { __update }
  })

const Organization = types
  .model({
    id: types.string,
    name: types.string,
    welcomeText: types.string,
    enableExposureText: types.string,
    recommendExposureText: types.string,
    notifyingOthersText: types.string,
    exposureInfoText: types.string,
    exposureAboutText: types.string,
    exposureDetailsText: types.string,
    exposureDetailsLearnText: types.string,
    verificationStartText: types.string,
    verificationIdentifierText: types.string,
    verificationIdentifierAboutText: types.string,
    verificationAdministrationDateText: types.string,
    verificationReviewText: types.string,
    verificationSharedText: types.string,
    verificationNotSharedText: types.string,
    diagnosisText: types.string,
    exposureText: types.string,
    currentPageOfMembers: types.array(User),
  })
  .actions((self) => {
    const __update = (updates) => {
      Object.keys(updates).forEach((key) => {
        if (self.hasOwnProperty(key)) self[key] = updates[key] // eslint-disable-line no-prototype-builtins
      })
      Logging.log('Updated Organization:')
      Logging.log(self)
    }

    const __setCurrentPageOfMembers = (pageOfMembers) => {
      self.currentPageOfMembers = cast(pageOfMembers)
    }

    return { __update, __setCurrentPageOfMembers }
  })

const Store = types.model({
  user: User,
  organization: Organization,
})

const defaultUser = {
  isSignedIn: false,
  email: '',
  isAdmin: false,
  isSuperAdmin: false,
  disabled: false,
  prefix: '',
  firstName: '',
  lastName: '',
  organizationID: '',
  imageBlob: null,
}

const defaultOrganization = {
  id: '',
  name: '',
  welcomeText: '',
  enableExposureText: '',
  recommendExposureText: '',
  notifyingOthersText: '',
  exposureInfoText: '',
  exposureAboutText: '',
  exposureDetailsText: '',
  exposureDetailsLearnText: '',
  verificationStartText: '',
  verificationIdentifierText: '',
  verificationIdentifierAboutText: '',
  verificationAdministrationDateText: '',
  verificationReviewText: '',
  verificationSharedText: '',
  verificationNotSharedText: '',
  diagnosisText: '',
  exposureText: '',
  currentPageOfMembers: [],
}

const defaultStore = {
  user: defaultUser,
  organization: defaultOrganization,
}

let initialStore = defaultStore

// Based on https://egghead.io/lessons/react-store-store-in-local-storage
if (localStorage.getItem('store')) {
  initialStore = JSON.parse(localStorage.getItem('store'))
}

const rootStore = Store.create({
  ...initialStore,
})

// Based on https://egghead.io/lessons/react-store-store-in-local-storage
onSnapshot(rootStore, (snapshot) => {
  localStorage.setItem('store', JSON.stringify(snapshot))
})

export { rootStore, defaultUser, defaultOrganization }
