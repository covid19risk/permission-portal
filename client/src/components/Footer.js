import React from 'react';
import { makeStyles } from '@material-ui/styles';
import Grid from '@material-ui/core/Grid';

const Footer = (props) => {
  return (
    <Grid container justify='center' alignItems='flex-end'>
      <Grid item>
        <div className={'footerContainer' + (props.branded ? ' branded' : '')}>
          <p id='copyright'>Â© 2020 Covid Watch. All rights reserved.</p>
          <a href='url'>Privacy Policy</a>
          <a href='url'>Terms of Use</a>
        </div>
      </Grid>
    </Grid>
  );
};

export default Footer;
