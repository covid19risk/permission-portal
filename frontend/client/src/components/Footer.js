import React from 'react'

const Footer = () => {
  const cwImg = require('../../assets/powered-by-cw.svg')

  return (
    <div id="footer">
      <div className="footerContainer">
        <div id="footer-img">
          <img src={cwImg}></img>
        </div>
        <div className="xs-text" id="footer-text">
          <p id="copyright">Â© 2020 Covid Watch. All rights reserved.</p>
          <a href="https://covidwatch.org/covid_watch_privacy_policy.pdf">Privacy Policy</a>
          <a href="url">Terms of Use</a>
          <a href="url">Support</a>
        </div>
      </div>
    </div>
  )
}

export default Footer
