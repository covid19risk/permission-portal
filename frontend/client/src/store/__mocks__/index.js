import { rootStore } from '../model'
import React from 'react'

const RootStoreContext = React.createContext()

const createStore = (WrappedComponent) => {
  rootStore.user.__update({ isSignedIn: true, isAdmin: false, isFirstTimeUser: false })
  return class extends React.Component {
    displayName = 'newStore'
    constructor(props) {
      super(props)
      this.data = rootStore
    }
    isSignInWithEmailLink = () => {}
    render() {
      return (
        <RootStoreContext.Provider value={this}>
          <WrappedComponent {...this.props} />
        </RootStoreContext.Provider>
      )
    }
  }
}

const withStore = (WrappedComponent) => {
  return class extends React.Component {
    displayName = 'storeConsumer'
    render() {
      return (
        <RootStoreContext.Consumer>
          {(store) => {
            return <WrappedComponent store={store} {...this.props} />
          }}
        </RootStoreContext.Consumer>
      )
    }
  }
}

export { createStore, withStore }
