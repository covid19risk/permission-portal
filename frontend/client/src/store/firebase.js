import app from 'firebase/app'
import 'firebase/auth'
import 'firebase/firestore'
import 'firebase/database'
import 'firebase/functions'
import 'firebase/analytics'
import * as firebaseConfigLocal from '../config/firebase.config.local'
import * as firebaseConfigDev from '../config/firebase.config.dev'
import * as firebaseConfigTest from '../config/firebase.config.test'
import * as firebaseConfigProd from '../config/firebase.config.prod'
import * as firebaseConfigStaging from '../config/firebase.config.staging'

var firebaseConfigMap = {
  development: firebaseConfigDev,
  test: firebaseConfigTest,
  production: firebaseConfigProd,
  local: firebaseConfigLocal,
  staging: firebaseConfigStaging,
}

const config = firebaseConfigMap[process.env ? process.env.NODE_ENV : 'development']

// declaring this as a global variable for use in index.js for Sentry environment labeling like "development", "staging", and "production"
window.env = process.env.NODE_ENV

app.initializeApp(config)
const auth = app.auth()
const SESSION = app.auth.Auth.Persistence.SESSION
const NONE = app.auth.Auth.Persistence.NONE
const db = app.firestore()
const analytics = app.analytics()
const createUserCallable = app.functions().httpsCallable('createUser')
const initiatePasswordRecoveryCallable = app.functions().httpsCallable('initiatePasswordRecovery')
const getVerificationCodeCallable = app.functions().httpsCallable('getVerificationCode')

export {
  auth,
  db,
  SESSION,
  NONE,
  analytics,
  createUserCallable,
  initiatePasswordRecoveryCallable,
  getVerificationCodeCallable,
}
