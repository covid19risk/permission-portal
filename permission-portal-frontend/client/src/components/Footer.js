import React from 'react'
import { useLocation } from 'react-router'

const Footer = () => {
  const cwImg = require('../../assets/powered-by-cw.svg')
  let location = useLocation()
  let style = {};
  let text = (
    <div className="footer">
      <p id="copyright">© 2020 Covid Watch. All rights reserved.</p>
      <a href="url">Privacy Policy</a>
      <a href="url">Terms of Use</a>
    </div>
  )
  
  if (location.pathname === '/') {
    // style remains empty object {}
  } else if (location.pathname === '/404') {
    // style remains empty object {}
    text = (
      <div></div>
    );
  } else {
    style = {
      minHeight: '100%',
      position: 'relative',
      background: 'url(' + cwImg + ')',
      backgroundRepeat: 'no-repeat',
      backgroundPositionX: '115px',
      backgroundPositionY: '36px',
    }
  }

  return (
    <div id="footer">
      <div
        className="footerContainer"
        style={style}
      >
        {text}
        {/* <p id="copyright">© 2020 Covid Watch. All rights reserved.</p>
        <a href="url">Privacy Policy</a>
        <a href="url">Terms of Use</a> */}
      </div>
    </div>
  )
}

export default Footer
