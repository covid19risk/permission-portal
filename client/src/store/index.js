import { types, flow } from 'mobx-state-tree'
import { firebase } from '../components/Firebase'

const User = types
  .model({
    uid: types.string,
    isAdmin: types.boolean,
    isSuperAdmin: types.boolean,
    prefix: types.string,
    firstName: types.string,
    lastName: types.string,
    role: types.maybeNull(types.string),
    organizationID: types.string
  });

const Organization = types
  .model({
    name: types.string,
    diagnosisText: types.string,
    exposureText: types.string
  })

const Store = types
  .model({
    user: types.maybeNull(User),
    organization: types.maybeNull(Organization)
  })
  .actions((self) => {
    const signIn = flow(function* (email, password) {
      try {
        yield firebase.doSignInWithEmailAndPassword(email, password)
        console.log('Successfully logged in')

        yield firebase.auth.currentUser.getIdTokenResult(true)
        const userDoc = yield firebase.getUserDocument(email)

        self.user = userDoc
        console.log("user: ", self.user)

        const orgDoc = yield firebase.getOrganizationDocument(self.user.organizationID)

        self.organization = orgDoc
        console.log("organization: ", self.organization)
      } catch (e) {
        console.log(e)
      }
    })

    const signOut = flow(function* () {
      try {
        yield firebase.doSignOut()
        console.log('Successfully logged out')
        self.user.isAdmin = false
      } catch (e) {
        console.log(e)
      }
    })

    const setOrganizationalBranding = flow(function* (diagnosisMessage, exposureMessage) {
      console.log('Setting organization branding');
      console.log(diagnosisMessage);
      console.log(exposureMessage);
      self.organization.diagnosisMessage = diagnosisMessage;
      self.organization.exposureText = exposureMessage;
      console.log('Succeeded');
    });

    return {
      signIn,
      signOut,
      setOrganizationalBranding
    }
  })

export default Store.create({
  user: null,
  organization: null
})
