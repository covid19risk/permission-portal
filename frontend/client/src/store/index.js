import React from 'react'
import { rootStore, defaultUser, defaultOrganization } from './model'
import Logging from '../util/logging'
import { auth, db, SESSION, createUserCallable, initiatePasswordRecoveryCallable } from './firebase'

const PAGE_SIZE = 15

const RootStoreContext = React.createContext()

const createStore = (WrappedComponent) => {
  return class extends React.Component {
    constructor(props) {
      super(props)
      auth.setPersistence(SESSION)
      this.data = rootStore
      this.__userDocumentListener = null
      this.__userImageListener = null
      this.__organizationDocumentListener = null
      this.__pageOfMembersListener = null
      this.__lastVisibleMember = null // pagination helper
      this.__firstVisibleMember = null // pagination helper
      this.__signedInWithEmailLink = false // firebase doesn't tell us this so we need to track it ourself
      this.__authStateListener = auth.onAuthStateChanged(async (user) => {
        if (user) {
          Logging.log('User signed in')
          // get user's document from the db and at the same time set up a listener to respond to document changes
          if (this.__userDocumentListener === null) {
            this.__userDocumentListener = db
              .collection('users')
              .doc(user.email)
              .onSnapshot((updatedUserDocumentSnapshot) => {
                // update the store with data from this document and set isSignedIn to true
                this.data.user.__update({
                  ...updatedUserDocumentSnapshot.data(),
                  email: updatedUserDocumentSnapshot.id,
                  isSignedIn: true,
                  signedInWithEmailLink: this.__signedInWithEmailLink,
                })
                // If DNE, set up organization listener within this callback,
                // since it relies on this.data.user.organizationID being set
                if (this.__organizationDocumentListener === null) {
                  this.__organizationDocumentListener = db
                    .collection('organizations')
                    .doc(this.data.user.organizationID)
                    .onSnapshot((updatedOrganizationDocumentSnapshot) => {
                      Logging.log('Remote organization document changed')
                      this.data.organization.__update({
                        ...updatedOrganizationDocumentSnapshot.data(),
                        id: updatedOrganizationDocumentSnapshot.id,
                        currentPage: this.data.organization.currentPage,
                      })
                      if (this.data.user.isAdmin && this.__pageOfMembersListener === null) {
                        var newListener = db
                          .collection('users')
                          .where('organizationID', '==', this.data.organization.id)
                          .orderBy('lastName')
                          .orderBy('firstName')
                          .limit(PAGE_SIZE)
                          .onSnapshot((pageOfMembersSnapshot) => {
                            this.__updatePageOfMembersOnSnapshot(pageOfMembersSnapshot.docs, newListener)
                          })
                      }
                    })
                }
              })
          }
          // Get and set up listener for user's image
          if (this.__userImageListener === null) {
            this.__userImageListener = db
              .collection('userImages')
              .doc(user.email)
              .onSnapshot((updatedUserImageDocumentSnapshot) => {
                if (updatedUserImageDocumentSnapshot.exists) {
                  this.data.user.__update({
                    imageBlob: updatedUserImageDocumentSnapshot.data().blob,
                  })
                }
              })
          }
        } else {
          Logging.log('User signed out')
          // signed out
          // reset to default state
          this.data.user.__update(defaultUser)
          this.data.organization.__update(defaultOrganization)

          // detach listeners
          if (this.__userDocumentListener !== null) {
            this.__userDocumentListener()
            this.__userDocumentListener = null
          }
          if (this.__userImageListener !== null) {
            this.__userImageListener()
            this.__userImageListener = null
          }
          if (this.__organizationDocumentListener !== null) {
            this.__organizationDocumentListener()
            this.__organizationDocumentListener = null
          }
          if (this.__pageOfMembersListener !== null) {
            this.__pageOfMembersListener()
            this.__pageOfMembersListener = null
          }
        }
      })
    }

    __updatePageOfMembersOnSnapshot(pageOfMembersSnapshotDocs, newListener) {
      if (pageOfMembersSnapshotDocs.length > 0) {
        if (this.__pageOfMembersListener != newListener) {
          this.__pageOfMembersListener = newListener
        }
        // Based on https://firebase.google.com/docs/firestore/query-data/query-cursors#paginate_a_query
        this.__firstVisibleMember = pageOfMembersSnapshotDocs[0]
        this.__lastVisibleMember = pageOfMembersSnapshotDocs[pageOfMembersSnapshotDocs.length - 1]
        this.data.organization.__setCurrentPageOfMembers(
          // See https://stackoverflow.com/a/24806827
          pageOfMembersSnapshotDocs.reduce((result, userDoc) => {
            if (userDoc.id !== this.data.user.email) {
              result.push({ ...userDoc.data(), email: userDoc.id })
            }
            return result
          }, [])
        )
      }
    }

    nextPageOfMembers() {
      var newListener = db
        .collection('users')
        .where('organizationID', '==', this.data.organization.id)
        .orderBy('lastName')
        .orderBy('firstName')
        .startAfter(this.__lastVisibleMember)
        .limit(PAGE_SIZE)
        .onSnapshot((pageOfMembersSnapshot) => {
          this.__updatePageOfMembersOnSnapshot(pageOfMembersSnapshot.docs, newListener)
        })
    }

    previousPageOfMembers() {
      var newListener = db
        .collection('users')
        .where('organizationID', '==', this.data.organization.id)
        .orderBy('lastName', 'desc')
        .orderBy('firstName', 'desc')
        .startAfter(this.__firstVisibleMember)
        .limit(PAGE_SIZE)
        .onSnapshot((pageOfMembersSnapshot) => {
          this.__updatePageOfMembersOnSnapshot(pageOfMembersSnapshot.docs.reverse(), newListener)
        })
    }

    async updateUser(updates) {
      try {
        await db.collection('users').doc(this.data.user.email).update(updates)
      } catch (err) {
        Logging.error('Error updating users', err)
      }
    }

    async updateUserImage(blob) {
      try {
        // .set() with { merge: true } so that if the document dne, it's created, otherwise its updated
        await db.collection('userImages').doc(this.data.user.email).set({ blob: blob }, { merge: true })
      } catch (err) {
        Logging.error('Error updating image', err)
      }
    }

    async updateOrganization(updates) {
      try {
        await db.collection('organizations').doc(this.data.organization.id).update(updates)
      } catch (err) {
        Logging.error('Error updating organization texts', err)
        throw err
      }
    }

    async signInWithEmailAndPassword(email, password) {
      this.__signedInWithEmailLink = false
      await auth.signInWithEmailAndPassword(email, password)
    }

    async signInWithEmailLink(email, link) {
      this.__signedInWithEmailLink = true
      return auth.signInWithEmailLink(email, link)
    }

    async signOut() {
      await auth.signOut()
    }

    async updatePassword(password) {
      await auth.currentUser.updatePassword(password)
    }

    async createUser(newUser) {
      try {
        const result = await createUserCallable(newUser)
        Logging.log(`Created new user: ${JSON.stringify(result.data)}`)
        return result.data
      } catch (err) {
        Logging.log(err)
        throw err
      }
    }

    async sendPasswordResetEmail(email) {
      try {
        await auth.sendPasswordResetEmail(email)
        return true
      } catch (err) {
        Logging.error(err)
        throw err
      }
    }

    async sendPasswordRecoveryEmail(email) {
      try {
        await initiatePasswordRecoveryCallable({ email: email })
        // Save the email locally so you don't need to ask the user for it again if they open the link on the same device.
        // See https://firebase.google.com/docs/auth/web/email-link-auth#send_an_authentication_link_to_the_users_email_address
        window.localStorage.setItem('emailForSignIn', email)
        return true
      } catch (err) {
        Logging.error(err)
        throw err
      }
    }

    async updateUserByEmail(email, updates) {
      try {
        await db.collection('users').doc(email).update(updates)
      } catch (err) {
        Logging.error(`Error updating user: ${email}`, err)
        throw err
      }
    }

    isSignInWithEmailLink(link) {
      return auth.isSignInWithEmailLink(link)
    }

    setPasswordResetCompletedInCurrentSession(val) {
      this.data.user.__update({ passwordResetCompletedInCurrentSession: val })
    }

    displayName = 'storeProvider'

    render() {
      return (
        <RootStoreContext.Provider value={this}>
          <WrappedComponent {...this.props} />
        </RootStoreContext.Provider>
      )
    }
  }
}

const withStore = (WrappedComponent) => {
  return class extends React.Component {
    displayName = 'storeConsumer'
    render() {
      return (
        <RootStoreContext.Consumer>
          {(store) => <WrappedComponent store={store} {...this.props} />}
        </RootStoreContext.Consumer>
      )
    }
  }
}

export { createStore, withStore, PAGE_SIZE }
