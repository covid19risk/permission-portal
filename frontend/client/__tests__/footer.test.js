import React from 'react'
import { mount } from 'enzyme'
import { rootStore } from '../src/store/model'
import Footer from '../src/components/Footer'
const https = require('https')

const wrapper = mount(<Footer store={{ data: rootStore }} />)

test('privacy policy link is valid', (done) => {
  function callback(res) {
    expect(res.statusCode).toBe(200)
    done()
  }

  https.get(wrapper.find('a').at(0).props().href, callback)
})

test('support link is valid', (done) => {
  function callback(res) {
    expect(res.statusCode).toBe(200)
    done()
  }

  https.get(wrapper.find('a').at(1).props().href, callback)
})
