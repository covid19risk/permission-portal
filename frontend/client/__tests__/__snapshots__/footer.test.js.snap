// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<Component
  store={
    Object {
      "data": Object {
        "organization": Object {
          "diagnosisText": "",
          "enableExposureText": "",
          "exposureAboutText": "",
          "exposureDetailsLearnText": "",
          "exposureDetailsText": "",
          "exposureInfoText": "",
          "exposureText": "",
          "id": "",
          "logoBlob": null,
          "members": Array [],
          "membersPage": 1,
          "name": "",
          "notifyingOthersText": "",
          "recommendExposureText": "",
          "verificationAdministrationDateText": "",
          "verificationIdentifierAboutText": "",
          "verificationIdentifierText": "",
          "verificationNotSharedText": "",
          "verificationReviewText": "",
          "verificationSharedText": "",
          "verificationStartText": "",
          "welcomeText": "",
        },
        "user": Object {
          "disabled": false,
          "email": "",
          "firstName": "",
          "imageBlob": null,
          "isAdmin": false,
          "isFirstTimeUser": true,
          "isSignedIn": false,
          "lastName": "",
          "organizationID": "",
          "passwordResetCompletedInCurrentSession": false,
          "passwordResetRequested": false,
          "prefix": "",
          "signedInWithEmailLink": false,
        },
      },
    }
  }
>
  <Memo(wrappedComponent)
    store={
      Object {
        "data": Object {
          "organization": Object {
            "diagnosisText": "",
            "enableExposureText": "",
            "exposureAboutText": "",
            "exposureDetailsLearnText": "",
            "exposureDetailsText": "",
            "exposureInfoText": "",
            "exposureText": "",
            "id": "",
            "logoBlob": null,
            "members": Array [],
            "membersPage": 1,
            "name": "",
            "notifyingOthersText": "",
            "recommendExposureText": "",
            "verificationAdministrationDateText": "",
            "verificationIdentifierAboutText": "",
            "verificationIdentifierText": "",
            "verificationNotSharedText": "",
            "verificationReviewText": "",
            "verificationSharedText": "",
            "verificationStartText": "",
            "welcomeText": "",
          },
          "user": Object {
            "disabled": false,
            "email": "",
            "firstName": "",
            "imageBlob": null,
            "isAdmin": false,
            "isFirstTimeUser": true,
            "isSignedIn": false,
            "lastName": "",
            "organizationID": "",
            "passwordResetCompletedInCurrentSession": false,
            "passwordResetRequested": false,
            "prefix": "",
            "signedInWithEmailLink": false,
          },
        },
      }
    }
  >
    <div
      id="footer"
    >
      <div
        className="footerContainer"
      >
        <div
          id="footer-img"
        >
          <img
            src={
              Object {
                "getCacheKey": [Function],
                "process": [Function],
              }
            }
          />
        </div>
        <div
          className="xs-text"
          id="footer-text"
        >
          <p
            id="copyright"
          >
            Â© 2020 Covid Watch. All rights reserved.
          </p>
          <a
            href="https://drive.google.com/file/d/1vSPtUEjSYQ0jWxYHlP8OzTzCAWgT9B0E/view?usp=sharing"
            rel="noreferrer"
            target="_blank"
          >
            Privacy Policy
          </a>
          <a
            href="https://covidwatch.zendesk.com/hc/en-us"
            rel="noreferrer"
            target="_blank"
          >
            Support
          </a>
        </div>
      </div>
    </div>
  </Memo(wrappedComponent)>
</Component>
`;
